diff --git a/node_modules/react-native-worklets/lib/module/initializers.js b/node_modules/react-native-worklets/lib/module/initializers.js
index 714206c..8fd85cc 100644
--- a/node_modules/react-native-worklets/lib/module/initializers.js
+++ b/node_modules/react-native-worklets/lib/module/initializers.js
@@ -128,7 +128,7 @@ function initializeWorkletRuntime() {
       const modules = require.getModules();
       // @ts-expect-error type not exposed by Metro
       const ReactNativeModuleId = require.resolveWeak('react-native');
-      const factory = function (_global, _$$_REQUIRE, _$$_IMPORT_DEFAULT, _$$_IMPORT_ALL, module, _exports, _dependencyMap) {
+      const factory = function (_global, _$$_REQUIRE, _$$_IMPORT_DEFAULT, _$$_IMPORT_ALL_2, module, _exports, _dependencyMap) {
         module.exports = new Proxy({}, {
           get: function get(_target, prop) {
             globalThis.console.warn(`You tried to import '${String(prop)}' from 'react-native' module on a Worklet Runtime. Using 'react-native' module on a Worklet Runtime is not allowed.`);
